src/ImageSharp/Formats/Bmp/BmpEncoderCore.cs

```diff
@@ -236,7 +235,7 @@ private BmpInfoHeader CreateBmpInfoHeader(int width, int height, int infoHeaderS
 
         if (this.infoHeaderType is BmpInfoHeaderType.WinVersion5 && metadata.IccProfile != null)
         {
-            infoHeader.ProfileSize = iccProfileData.Length;
+            infoHeader.ProfileSize = iccProfileData!.Length;
```

Comment:
It's weird that we have to do this given the null check conditional.

Reply:
Read the null check carefully. We check the null on another object

Reply:
Ah yes... 

This reads like a bad method then. Why are we even passing the metadata as a parameter when we can pass the byte[]? and do the null check on that?